src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -60,26 +66,86 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
 
                 if (!string.IsNullOrWhiteSpace(proxyHelper.ProxyBypass))
                 {
+                    int idx = 0;
+                    int start = 0;
+                    string tmp;
+
                     // Process bypass list for manual setting.
-                    string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(proxyHelper.ProxyBypass.Length / 5); // Best guess based on string length.
 
-                    foreach (string value in list)
+                    while (idx < proxyHelper.ProxyBypass.Length)
                     {
-                        string tmp = value.Trim();
-                        if (tmp == "<local>")
+                        // Strip leading spaces and scheme if any.
+                        while (idx < proxyHelper.ProxyBypass.Length && proxyHelper.ProxyBypass[idx] == ' ') { idx += 1; };
+                        if (String.Compare(proxyHelper.ProxyBypass, idx, "http://", 0, 7, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            idx += 7;
+                        }
+                        else if (String.Compare(proxyHelper.ProxyBypass, idx, "https://", 0, 8, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            idx += 8;
+                        }
+
+                        start = idx;
+                        while (idx < proxyHelper.ProxyBypass.Length && proxyHelper.ProxyBypass[idx] != ' ' && proxyHelper.ProxyBypass[idx] != ';') {idx += 1; };
+
+                        if (idx == start)
+                        {
+                            // Empty string.
+                            tmp = null;
+                        }
+                        else if (String.Compare(proxyHelper.ProxyBypass, start, "<local>", 0, 7, StringComparison.OrdinalIgnoreCase) == 0)
                         {
                             _bypassLocal = true;
+                            tmp = null;
+                        }
+                        else
+                        {
+                            tmp = proxyHelper.ProxyBypass.Substring(start, idx-start);
+                        }
+
+                        // Skip trailing characters if any.
+                        if (idx < proxyHelper.ProxyBypass.Length && proxyHelper.ProxyBypass[idx] != ';')
+                        {
+                            // Got stopped at space.
```

Comment:
Did you mean colon rather than space?

Reply:
no, it does not matter. we just want to skip till next delimiter e.g. ';'